scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -354,6 +354,10 @@ def monitor_serial(self, ser, halt_event, harness):
             # from the test.
             harness.capture_coverage = True
 
+        # Wait for serial connection
```

Comment:
could you please elaborate the comment: why it needs to wait for serial connection here ?
and how the reading thread is expected to get out of this waiting loop when the port remains closed ?